### PR TITLE
handle atttypmod being correct for decimals from v22.1

### PR DESCRIPTION
**Pull request by otan**

### CODE REVIEW

File: lib/active_record/connection_adapters/cockroachdb_adapter.rb [hunk 1]
```diff
@@ -291,21 +297,23 @@ def initialize_type_map(m = type_map)
...
-            # If fmod is -1, that means that precision is defined but not
-            # scale, or neither is defined.
-            if fmod && fmod == -1 && !precision.nil?
-              # Below comment is from ActiveRecord
-              # FIXME: Remove this class, and the second argument to
-              # lookups on PG
-              Type::DecimalWithoutScale.new(precision: precision)
-            else
-              OID::Decimal.new(precision: precision, scale: scale)
+          if needs_numeric_override?
```
Comment (rafiss) [inline lib/active_record/connection_adapters/cockroachdb_adapter.rb:302]: it seems that this is getting called before `@crdb_version` is defined
```
20:20:30   /home/agent/activerecord-cockroachdb-adapter/lib/active_record/connection_adapters/cockroachdb_adapter.rb:210:in `needs_numeric_override?': undefined method `<' for nil:NilClass (NoMethodError)

20:20:30     from /home/agent/activerecord-cockroachdb-adapter/lib/active_record/connection_adapters/cockroachdb_adapter.rb:302:in `initialize_type_map'

20:20:30     from /home/agent/.gems/bundler/gems/rails-f90e6d14b9c8/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:261:in `initialize'

20:20:30     from /home/agent/activerecord-cockroachdb-adapter/lib/active_record/connection_adapters/cockroachdb_adapter.rb:233:in `initialize'

20:20:30     from /home/agent/activerecord-cockroachdb-adapter/lib/active_record/connection_adapters/cockroachdb_adapter.rb:40:in `new'

20:20:30     from /home/agent/activerecord-cockroachdb-adapter/lib/active_record/connection_adapters/cockroachdb_adapter.rb:40:in `cockroachdb_connection'
```

**Review by rafiss — CHANGES_REQUESTED** [review]

File: lib/active_record/connection_adapters/cockroachdb_adapter.rb (hunk 1 above)
Comment (otan) [reply lib/active_record/connection_adapters/cockroachdb_adapter.rb:302]: not sure how to fix this then :\

Comment (otan) [reply lib/active_record/connection_adapters/cockroachdb_adapter.rb:302]: `initialize_type_map` is called in `super.initialize`, but we need to be initialized before we can issue the query to get the crdb version

Comment (rafiss) [reply lib/active_record/connection_adapters/cockroachdb_adapter.rb:302]: what if we change the condition below from
```
if fmod && fmod == -1 && !precision.nil?
```
to
```
if fmod && ((fmod == -1 && !precision.nil?) || (fmod - 4 & 0xffff).zero?)
```

ref: https://github.com/rails/rails/blob/87d4d0f4126f64d991d40a1827de50935ddfdbff/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb#L569

Comment (otan) [reply lib/active_record/connection_adapters/cockroachdb_adapter.rb:302]: also these kind of checks are super broken for mixed version clusters no? is this initalize method global?

Comment (otan) [reply lib/active_record/connection_adapters/cockroachdb_adapter.rb]: ah that works

Comment (rafiss) [reply lib/active_record/connection_adapters/cockroachdb_adapter.rb]: yeah good point about mixed version clusters :(

i believe this initialize happens once per connection

**Review by rafiss — APPROVED** [review]
ty!

**Pull request merged**